drivers/xen/gnttab.c

```diff
@@ -86,7 +87,7 @@ static void gnttab_grant_permit_access(grant_ref_t gref, domid_t domid,
 	gnttab.table[gref].frame = gfn;
 	gnttab.table[gref].domid = domid;
 	/* Need to be sure that gfn and domid will be set before flags */
-	__DMB();
+	barrier_data_memory_fence_full();
```

Comment:
I believe, we need `barrier_data_synchronization_fence_full` in this case, as this part synchronizes access to grant references tables between Zephyr and Hypervisor. I am not sure that GCC's idea of synchronization is valid for inter-VM communications, so I believe we need proper `dsb sy` there.

Reply:
idem as in my previous comment.